usb.c

```diff
@@ -1200,13 +1200,70 @@ static int usb_fill_context_info(struct iio_context_info *info,
 	return 0;
 }
 
-int usb_context_scan(struct iio_scan_result *scan_result)
+#define SCAN_WILDCARD "*"
+#define SCAN_DELIMITER ','
+
+static int parse_value(const char *str)
+{
+	uint32_t val;
+	char *ptr;
+
+	/* if it is null or '*' accept any */
+	if (str == NULL || !strncmp(str, SCAN_WILDCARD, sizeof(SCAN_WILDCARD)))
+		return 0;
```

Comment:
I believe you only want to check for the wildcard character for PID, since as we talked about before, it does not make sense to use it for the VID, and in the case where we want to scan every USB device then the backend string can be NULL.

Reply:
Yes - that's the way it works:
```
rgetz@brain:~/github/libiio/build$ ./tests/iio_info -S usb=*:
Library version: 0.23 (git tag: 7b2c94a)
Compiled with backends: local xml ip usb
Scanning for IIO contexts failed: No such device (19)
```

but we use the same parsing for pid and vid...